solver/src/solver.rs

```diff
@@ -0,0 +1,8 @@
+use crate::settlement::Settlement;
+use anyhow::Result;
+use model::order::Order;
+
+#[async_trait::async_trait]
+pub trait Solver {
+    async fn solve(&self, orders: Vec<Order>) -> Result<Option<Settlement>>;
+}
```

Comment:
nit: docs for this public trait (in particular the semantics of the return type) would be nice.